hrcase/CRM/HRCase/Upgrader.php

```diff
@@ -471,4 +485,62 @@ private function removeCaseTypesWithData($caseTypes) {
     }
   }
 
+  /**
+   * @param $component
+   * @param array $types
+   */
+  private function installActivityTypes($component, array $types) {
```

Comment:
so if you followed my suggestion above you can remove both **installActivityTypes** & **fetchActivityTypeParams** methods .. by the way , civicrm understands component and option group names , so you can just pass the component name and option group name without the need to fetch their IDs first , so you can simply just use : 

```php
    civicrm_api3('OptionValue', 'create', array(
 +        'sequential' => 1,
 +        'option_group_id' => 'activity_type',
 +        'component_id' => 'CiviTask',
 +        'label' => $type,
 +        'name' => $type,
 +      ));
```

you can use API explorer to create you API calls and better understanding for how things work